Tests/Unit/Controller/Admin/ModuleSupportTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use Wirecard\Oxid\Controller\Admin\ModuleSupport;
+use Wirecard\Oxid\Core\Helper;
+
+class ModuleSupportTest extends \Wirecard\Test\WdUnitTestCase
+{
+    /**
+     * @var ModuleSupport
+     */
+    private $_moduleSupport;
+
+    protected function setUp()
+    {
+        $this->_moduleSupport = oxNew(ModuleSupport::class);
+        parent::setUp();
+    }
+
+    protected function failOnLoggedExceptions()
+    {
+        // don't fail on logged exception -> mailer is not instantiated
+        $this->exceptionLogHelper->clearExceptionLogFile();
+    }
+
+    public function testRender()
+    {
+        $this->_moduleSupport->render();
+        $this->assertArrayHasKey('oxid', $this->_moduleSupport->getViewData());
+        $this->assertArrayHasKey('contactEmail', $this->_moduleSupport->getViewData());
+        $this->assertArrayHasKey('defaultEmail', $this->_moduleSupport->getViewData());
+        $this->assertArrayHasKey('isOurModule', $this->_moduleSupport->getViewData());
+    }
+
+    public function testSendSupportEmailActionWithoutParams()
+    {
+        $this->_moduleSupport->sendSupportEmailAction();
+        $this->assertArrayHasKey('alertMessage', $this->_moduleSupport->getViewData());
+        $this->assertArrayHasKey('alertType', $this->_moduleSupport->getViewData());
+    }
+
+    /**
+     * @dataProvider testSendSupportEmailActionWithParamsProvider
+     */
+    public function testSendSupportEmailActionWithParams($sText, $sFromEmail, $sReplyEmail, $sErrorText)
+    {
+        $this->_moduleSupport->setEditObjectId(Helper::MODULE_ID);
+        $_POST['module_support_text'] = $sText;
+        $_POST['module_support_email_from'] = $sFromEmail;
+        $_POST['module_support_email_reply'] = $sReplyEmail;
+
+        $this->_moduleSupport->sendSupportEmailAction();
+
+        $this->assertEquals($sErrorText, $this->_moduleSupport->getViewData()['alertMessage']);
+    }
+
+    public function testSendSupportEmailActionWithParamsProvider()
+    {
+        return [
+            // this will end up in a flaky test depending on the email settings of the test server
+            // 'correct params' => ['support text', 'from@email.com', 'reply@email.com', Helper::translate('wd_support_send_error')],
```

Comment:
👍

Reply:
because of the comment above. We'll have a flaky test depending on the shops config of the email server. In our containers email is not supported. In the OXID testing VM, email is enabled and will result in a different outcome.

Reply:
Dose it make sense to, add a global variable which you can set pre environment if the email tests are run?

Reply:
Hmm, nice idea, but how will you communicate that to OXID, when they run the tests, that they have to set some values? For the certification I mean.

Reply:
We could also check for an ENV variable that is only present in our Docker setup. If it is set, we skip the test. What do you think?